File: C/minimum_absolute_difference_in_BST.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
int doMinimumDifference(struct TreeNode* root, int* pre, int* result){
    if(!root) return 0;
    
    if(root->left) *result = doMinimumDifference(root->left, pre, result);
    
    if(*pre >= 0) *result = fmin(*result, abs(*pre - root->val));
    *pre = root->val;
        
    if(root->right) *result = doMinimumDifference(root->right, pre, result);
    
    return *result;
}

int getMinimumDifference(struct TreeNode* root){
    if(!root) return 0;
    
    int result = INT_MAX, pre = -1;
    
    doMinimumDifference(root, &pre, &result);
        
    return result;
}

File: Python3/minimum_absolute_difference_in_BST.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def getMinimumDifference(self, root: TreeNode) -> int:
        if not root: return 0
        
        stack = []
        
        def dfs(root):
            if not root: return
            stack.append(root.val)
            dfs(root.left)
            dfs(root.right)
        dfs(root)
        
        stack.sort()
        result = stack[1] - stack[0]
        for i in range(1, len(stack)):
            diff = stack[i] - stack[i-1]
            if diff < result:
                result = diff
        
        return result
